src/zkevm_specs/evm/execution/chainid.py

```diff
@@ -0,0 +1,21 @@
+from zkevm_specs.util.param import N_BYTES_WORD
+from ..instruction import Instruction, Transition
+from ..table import BlockContextFieldTag
+from ..opcode import Opcode
+
+
+def chainid(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.CHAINID)
+
+    # check block table for CHAINID value
+    instruction.constrain_equal(
+        instruction.stack_push(), instruction.block_context_lookup(BlockContextFieldTag.ChainId)
+    )
+
+    instruction.step_state_transition_in_same_context(
+        opcode,
+        rw_counter=Transition.delta(1),
```

Comment:
```suggestion
        rw_counter=Transition.delta(2),
```

gc + 2 (1 stack write + 1 Call Context read) ?

Reply:
As discussed in private. This does not apply as the BlockContext lookups do not count as RW operation in the bus-mapping :)

Reply:
Thanks for clarifying, @CPerezz !